lib/auth/result.rb

```diff
@@ -52,6 +52,13 @@ def email
     @email&.downcase
   end
 
+  def email_valid=(val)
+    if val == 'false'
+      raise ArgumentError, "email_valid is a 'false' string. Convert to boolean before creating Auth::Result"
+    end
+    @email_valid = !!val
```

Comment:
Is `"false"` the only invalid input this can receive? If not, perhaps allowing a certain set of values and raising on anything else would be better? 🤷

Reply:
I made this PR because someone had configured the `discourse-oauth2-basic` plugin to use use the actual email address as the `email_verified` value.

Ideally, only `true` and `false` would be valid values. My worry is that enforcing that will break some auth plugins... maybe we should just rip off the band-aid and identify all the issues.

Reply:
I'd either go all in and let any invalid auth plugins burn 😉 or accept all input and try to auto-correct (i.e. instead of raising on `"false"` treat it as `false`)

One is more "correct" but provides you with with potential work for the years to come 😜

Reply:
Let's just go all-in. Auth, and email validation, are security-critical things so I think it's better for things to fail than have weird edge cases. Will wait until after the holidays to merge though - we don't want to accidentally make extra work for ourselves over the next few days.